<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="en"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="../jacoco-resources/report.css" type="text/css"/><link rel="shortcut icon" href="../jacoco-resources/report.gif" type="image/gif"/><title>CancelDebuggingListenableFutureTest.java</title><link rel="stylesheet" href="../jacoco-resources/prettify.css" type="text/css"/><script type="text/javascript" src="../jacoco-resources/prettify.js"></script></head><body onload="window['PR_TAB_WIDTH']=4;prettyPrint()"><div class="breadcrumb" id="breadcrumb"><span class="info"><a href="../jacoco-sessions.html" class="el_session">Sessions</a></span><a href="../index.html" class="el_report">threadly$All_in_threadly_test.exec</a> &gt; <a href="index.source.html" class="el_package">org.threadly.concurrent.future</a> &gt; <span class="el_source">CancelDebuggingListenableFutureTest.java</span></div><h1>CancelDebuggingListenableFutureTest.java</h1><pre class="source lang-java linenums">package org.threadly.concurrent.future;

import static org.junit.Assert.*;

import java.util.concurrent.CancellationException;
import java.util.concurrent.ExecutionException;

import org.junit.Test;
import org.threadly.concurrent.future.CancelDebuggingListenableFuture.FutureProcessingStack;

@SuppressWarnings(&quot;javadoc&quot;)
<span class="fc" id="L12">public class CancelDebuggingListenableFutureTest {</span>
  @Test
  public void notStartedTest() throws InterruptedException, ExecutionException {
<span class="fc" id="L15">    SettableListenableFuture&lt;Object&gt; slf = new SettableListenableFuture&lt;&gt;();</span>
<span class="fc" id="L16">    CancelDebuggingListenableFuture&lt;Object&gt; debugFuture = new CancelDebuggingListenableFuture&lt;&gt;(slf);</span>
    
<span class="fc" id="L18">    assertTrue(debugFuture.cancel(false));</span>
    
    try {
<span class="nc" id="L21">      debugFuture.get();</span>
<span class="nc" id="L22">      fail(&quot;Exception should have thrown&quot;);</span>
<span class="fc" id="L23">    } catch (CancellationException e) {</span>
      // expected
<span class="fc" id="L25">      assertNull(e.getCause());</span>
<span class="nc" id="L26">    }</span>
<span class="fc" id="L27">  }</span>
  

  @Test
  public void withRunningStackTest() throws InterruptedException, ExecutionException {
<span class="fc" id="L32">    SettableListenableFuture&lt;Object&gt; slf = new SettableListenableFuture&lt;&gt;();</span>
<span class="fc" id="L33">    slf.setRunningThread(Thread.currentThread());</span>
<span class="fc" id="L34">    CancelDebuggingListenableFuture&lt;Object&gt; debugFuture = new CancelDebuggingListenableFuture&lt;&gt;(slf);</span>
    
<span class="fc" id="L36">    assertTrue(debugFuture.cancel(false));</span>
    
    try {
<span class="nc" id="L39">      debugFuture.get();</span>
<span class="nc" id="L40">      fail(&quot;Exception should have thrown&quot;);</span>
<span class="fc" id="L41">    } catch (CancellationException e) {</span>
      // expected
<span class="fc" id="L43">      assertNotNull(e.getCause());</span>
<span class="fc" id="L44">      assertTrue(e.getCause() instanceof FutureProcessingStack);</span>
<span class="fc" id="L45">      assertEquals(this.getClass().getName(), e.getCause().getStackTrace()[3].getClassName());</span>
<span class="nc" id="L46">    }</span>
<span class="fc" id="L47">  }</span>
}
</pre><div class="footer"><span class="right">Created with <a href="http://www.jacoco.org/jacoco">JaCoCo</a> 0.8.1.201803210924</span></div></body></html>